tests/SideBySide/AppConfig.cs

```diff
@@ -74,19 +73,8 @@ public static MySqlConnectionStringBuilder CreateSha256ConnectionStringBuilder()
 		// tests can run much slower in CI environments
 		public static int TimeoutDelayFactor { get; } = (Environment.GetEnvironmentVariable("APPVEYOR") == "True" || Environment.GetEnvironmentVariable("TRAVIS") == "true") ? 6 : 1;
 
-		private static string GetCodeRootPath()
-		{
-#if NET46
-			var currentAssembly = Assembly.GetExecutingAssembly();
-#else
-			var currentAssembly = typeof(AppConfig).GetTypeInfo().Assembly;
-#endif
-			var directory = new Uri(currentAssembly.CodeBase).LocalPath;
-			while (!string.Equals(Path.GetFileName(directory), "MySqlConnector", StringComparison.OrdinalIgnoreCase))
-				directory = Path.GetDirectoryName(directory);
-			return directory;
-		}
-
-		private static string ExpandVariables(string value) => value?.Replace("%TESTDATA%", TestDataPath);
+		private static string ExpandVariables(string value, string replacement) => value?.Replace("%TESTDATA%", replacement);
+		private static string ExpandLocalVariables(string value) => ExpandVariables(value, TestDataPath);
+		private static string ExpandRemoteVariables(string value) => ExpandVariables(value, RemoteTestDataPath);
```

Comment:
If you need to use two variables (and I'm not sure why that's necessary), this should be:

```
private static string ExpandVariables(string value) =>
value?.Replace("%TESTDATA%", TestDataPath)?.Replace("%REMOTETESTDATA%", RemoteTestDataPath);
```

I.e., use two variables; don't overload the meaning of one.